src/ase/cw/view/ServerFrame.java

```diff
@@ -72,47 +81,63 @@ public int getServerId() {
 
         if (e.getSource() == breakButton) {
-            logButtonPress(breakButton);
-            breakButton.setEnabled(false);
-            restartButton.setEnabled(true);
-            oc.pauseOrderProcess(this.serverId);
+            if(pausable!=null) {
+                logButtonPress(breakButton);
+                breakButton.setEnabled(false);
+                restartButton.setEnabled(true);
+                pausable.pause();
+            }
+
         }
 
         if (e.getSource() == restartButton) {
-            logButtonPress(restartButton);
-            restartButton.setEnabled(false);
-            breakButton.setEnabled(true);
-            oc.restartOrderProcess(this.serverId);
+            if(pausable!=null){
+                logButtonPress(restartButton);
+                restartButton.setEnabled(false);
+                breakButton.setEnabled(true);
+                pausable.unPause();
+            }
         }
     }
 
     @Override
     public void updateView(OrderConsumer server, Order order) {
-        this.textArea.setText(
-                "Status: "+(server.getStatus()
-                        +"\nOrder: "+order.getCustomerId()+ (order.isPriorityOrder() ? " (priority)" : "")
-                        +"\nItems: "+order.getOrderItems().size()
-                        +"\nSubtotal: £"+order.getBill().getSubtotal()
-                        +"\nTotal: £"+order.getBill().getTotal()
-                ));
+
+        this.statusArea.setText("Status: " + (server.getStatus()));
+        if(server.getStatus()== ServerStatusEnum.ServerStatus.STOPPED){
+            this.dispose();
+        } else if (server.getStatus() == ServerStatusEnum.ServerStatus.BUSY && order != null) {
+            this.textArea.setText(
+                             "Order: " + order.getCustomerId()
+                            + "\nItems: " + order.getOrderItems().size()
+                            + "\nSubtotal: £" + order.getBill().getSubtotal()
+                            + "\nTotal: £" + order.getBill().getTotal()
+                    );
+        } else if(server.getStatus()== ServerStatusEnum.ServerStatus.FREE || server.getStatus()== ServerStatusEnum.ServerStatus.PAUSED){
+            this.textArea.setText("");
+        } else {
+
         }
+    }
 
     @Override
     public void closeFrame() {
-        dispose();
+        this.dispose();
     }
 
+
     @Override
     public void updateView(OrderConsumer server) {
-        this.textArea.setText(
-                "Status: "+(server.getStatus()
-                ));
+        updateView(server,null);
     }
 
+
+
     private class exitButtonPress extends WindowAdapter {
         public void windowClosing(WindowEvent evt) {
             Log.getLogger().log("GUI: ServerFrame "+serverId+" window close button pressed.");
-            oc.removeServer(oc.getServerById(serverId));
+            pausable.stop();
+            //TODO: stop server thread and dispose of server frame after current order is processed
```

Comment:
this is closing frame immediately and instead it should wait until server is finished. you can close all ServerFrames but the servers still process orders which makes no sense as you can't see it anywhere

Reply:
fixed,
Added a setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE) line. 
Now if you press exit, the server will stop and the window will be still visible. As soon as the server is in state "STOPPED" the window closes.

Reply:
it's still an issue with the design of order processing - when there is only one item in order the frame immediately closes. In addition to this when server is PAUSED and you want to close the window then the state is changed to TO_BE_PAUSED and the window can never be closed. I can set this up as an issue for the final Sprint if you prefer

Reply:
>it's still an issue with the design of order processing - when there is only one item in order the frame immediately closes.  
Yes, but this issue existed before too. Never mind I fixed it in the new Commit.

> In addition to this when server is PAUSED and you want to close the window then the state is changed to TO_BE_PAUSED and the window can never be closed. 
Fixed in the new Commit